grudge/reductions.py

```diff
@@ -178,11 +179,13 @@ def nodal_min(dcoll: DiscretizationCollection, dd, vec) -> float:
 
     # NOTE: Don't move this
     from mpi4py import MPI
+    actx = vec.array_context
 
-    return comm.allreduce(nodal_min_loc(dcoll, dd, vec), op=MPI.MIN)
+    return actx.from_numpy(
+        comm.allreduce(actx.to_numpy(nodal_min_loc(dcoll, dd, vec)), op=MPI.MIN))
 
 
-def nodal_min_loc(dcoll: DiscretizationCollection, dd, vec) -> float:
+def nodal_min_loc(dcoll: DiscretizationCollection, dd, vec) -> Any:
```

Comment:
In the case here, I think we should just document the return behavior in the docstring rather than annotate with `Any`. For this function (and similar), I think it'll be helpful to mention that the return type of these routines will depend on `force_device_scalars`

Reply:
Just to add to this, the reason I suggested adding `Any` is that `mypy` won't type check this function at all if it doesn't have a return type, but it's not an issue at the moment, I guess.

Completely agree it's useless from a documentation point of view.

Reply:
Also, maybe mentioning that `force_device_scalars` thing is not the best, since it's just a `PyOpenCLArrayContext` thing that's supposed to be deprecated already?